ethclient/ethclient.go

```diff
@@ -76,6 +76,21 @@ type rpcBlock struct {
 	UncleHashes  []common.Hash    `json:"uncles"`
 }
 
+func (ec *Client) BlockNumber(ctx context.Context) (big.Int, error) {
+	var numberstr string
```

Comment:
I still don't understand why we are using a string? Can the response not be decoded directly into `big.Int`?